### PR TITLE
fix(apim): fix cli parameter validation

### PR DESCRIPTION
**Pull request by XiaofuHuang**
Add answer validation for CLI platform until the question model validation is ready.

https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/9823734

### CODE REVIEW

File: packages/fx-core/src/plugins/resource/apim/index.ts [hunk 1]
```diff
@@ -118,7 +118,7 @@ async function _postProvision(ctx: PluginContext, progressBar: ProgressBar): Pro
     const aadConfig = new AadPluginConfig(ctx.configOfOtherPlugins);
 
     const apimManager = await Factory.buildApimManager(ctx, solutionConfig);
-    const aadManager = await Factory.buildAadManager(ctx, );
+    const aadManager = await Factory.buildAadManager(ctx,);
```
Comment (dooriya) [inline packages/fx-core/src/plugins/resource/apim/index.ts:121]: `buildAadManager(ctx,)` => `buildAadManager(ctx)`

Comment (XiaofuHuang) [reply packages/fx-core/src/plugins/resource/apim/index.ts]: got it.

File: packages/fx-core/src/plugins/resource/apim/index.ts [hunk 2]
```diff
@@ -136,6 +136,17 @@ async function _deploy(ctx: PluginContext, progressBar: ProgressBar): Promise<vo
     const apimConfig = new ApimPluginConfig(ctx.config);
     const functionConfig = new FunctionPluginConfig(ctx.configOfOtherPlugins);
     const answer = buildAnswer(ctx);
+
+    // TODO: delete the following logic after cli question model fix undefined / empty string validation bug
```
[the remark this answers is not in the document]
Comment (XiaofuHuang) [reply packages/fx-core/src/plugins/resource/apim/index.ts]: The bug links have been added.

File: packages/fx-core/src/plugins/resource/apim/constants.ts [hunk 3]
```diff
@@ -140,6 +140,12 @@ export class ValidationConstants {
         regex: /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i,
         message: "The value should be a GUID."
     }
+
+    public static readonly CLI = {
+        invalidOptionMessage: (optionName: string) => `The value of option '--${optionName}' is invalid.`,
+        emptyOptionMessage: (optionName: string) => `Option '--${optionName}' is required. Set the value of '--${optionName}'`,
+        overrideOptionMessage: (optionName: string) => `Option '--${optionName}' cannot be override. Remove option '--${optionName}'`,
```
Comment (dooriya) [inline packages/fx-core/src/plugins/resource/apim/constants.ts:147]: override -> overrode?

Comment (XiaofuHuang) [reply packages/fx-core/src/plugins/resource/apim/constants.ts]: Changed it to overridden

**Review by dooriya — APPROVED** [review]

**Pull request merged**